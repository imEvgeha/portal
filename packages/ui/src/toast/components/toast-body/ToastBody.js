import React from 'react';
import PropTypes from 'prop-types';
import './ToastBody.scss';

const ToastBody = ({severity, summary, detail, children}) => {
    const icons = {success: 'pi-check', warn: 'pi-exclamation-triangle', error: 'pi-times'};
    return (
        <div className="flex flex-column nexus-c-toast__container">
            <div className={`row type-${severity}`}>
                <div className="text-center col-2 nexus-c-toast__icon-container">
                    <i className={`pi ${icons[severity]} pi-fluid nexus-c-toast__icon`} />
                </div>
                <div className="col-10">
                    <div className="row g-0 nexus-c-toast__row-without-padding">
                        <span className="nexus-c-toast__summary-title">{summary}</span>
                    </div>
                    <div className="row g-0 nexus-c-toast__row-without-padding">
                        <span className="nexus-c-toast__detail-title">{detail}</span>
                        {children}
                    </div>
                </div>
            </div>
        </div>
    );
};

ToastBody.propTypes = {
    severity: PropTypes.string,
    summary: PropTypes.string,
    detail: PropTypes.string,
};

ToastBody.defaultProps = {
    severity: 'success',
    summary: '',
    detail: '',
};

export default ToastBody;
