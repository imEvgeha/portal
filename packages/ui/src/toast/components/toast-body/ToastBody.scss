@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/mixins';
@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/colors';
@import '~@vubiquity-nexus/portal-styles/scss/color-palette';

.nexus-c-toast__container {
    flex: 1;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    .nexus-c-toast__summary-title {
        font-size: 13px;
        font-weight: bold;
        color: palette(black, 100);
    }

    .nexus-c-toast__detail-title {
        font-size: 10px;
        font-weight: bold;
        color: palette(black, 100);
    }
}

.nexus-c-toast__icon-container {
    display: flex;
    justify-content: center;
    align-items: center;

    .nexus-c-toast__icon {
        font-size: 2rem;
    }
}
