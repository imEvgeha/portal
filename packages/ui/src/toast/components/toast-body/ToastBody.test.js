import React from 'react';
import {shallow} from 'enzyme';
import ToastBody from './ToastBody';

describe('WarningToastWithConfirmation', () => {
    let wrapper = null;

    beforeAll(() => {
        wrapper = shallow(<ToastBody />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render main container', () => {
        expect(wrapper.find('.nexus-c-toast__container')).toHaveLength(1);
    });

    it('should render icon', () => {
        expect(wrapper.find('.nexus-c-toast__icon-container')).toHaveLength(1);
    });
});
