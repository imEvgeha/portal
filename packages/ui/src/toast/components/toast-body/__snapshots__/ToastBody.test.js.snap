// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningToastWithConfirmation should match snapshot 1`] = `
<div
  className="flex flex-column nexus-c-toast__container"
>
  <div
    className="row type-success"
  >
    <div
      className="text-center col-2 nexus-c-toast__icon-container"
    >
      <i
        className="pi pi-check pi-fluid nexus-c-toast__icon"
      />
    </div>
    <div
      className="col-10"
    >
      <div
        className="row g-0 nexus-c-toast__row-without-padding"
      >
        <span
          className="nexus-c-toast__summary-title"
        />
      </div>
      <div
        className="row g-0 nexus-c-toast__row-without-padding"
      >
        <span
          className="nexus-c-toast__detail-title"
        />
      </div>
    </div>
  </div>
</div>
`;
